### PR TITLE
MG-111- Modify Chart modals

### LINKED ISSUE #111: [FEATURE] Modify Chart modals
### Is there an existing feature request for this?

- [X] I have searched the existing feature requests

### Is your feature request related to a problem? Please describe.

Modify chart modals to sync with the individual chart requirements

### Describe the feature you are requesting, as well as the possible use case(s) for it.

We need to be able to fetch data from Magistrala and be able to populate the charts with this data

### Indicate the importance of this feature to you.

- [X] Must-have
- [ ] Should-have
- [ ] Nice-to-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by 1998-felix**
# What type of PR is this?

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Dependency Update

## What does this do?
Modify chart modals to sync with individual charts

## Which issue(s) does this PR fix/relate to?

- Resolves #111 

## Have you included tests for your changes?

- [ ] Yes
- [x] No, and this is why: 

## Did you document any new/modified functionality?

- [ ] Yes
- [x] No, and this is why:

### CODE REVIEW

File: ui/web/templates/charts/horizontalbarchartmodal.html [hunk 1]
```diff
@@ -44,6 +44,16 @@ <h5 class="modal-title" id="horizontalBarChartModalLabel">Horizontal bar chart</
                 placeholder="Enter channel ID"
               />
             </div>
+            <div class="mb-3">
```
Comment (ianmuchyri) [inline ui/web/templates/charts/horizontalbarchartmodal.html:47]: There is no appearance tab for this modal

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: ui/web/templates/charts/linechartmodal.html [hunk 2]
```diff
@@ -96,21 +237,44 @@ <h5 class="modal-title" id="lineChartModalLabel">Time series line charts</h5>
       }
       form.reset();
 
+      console.log(formData);
```
Comment (ianmuchyri) [inline ui/web/templates/charts/linechartmodal.html:240]: remove log

File: ui/web/templates/charts/piechartmodal.html [hunk 3]
```diff
@@ -45,23 +45,34 @@ <h5 class="modal-title" id="pieChartModalLabel">Pie chart</h5>
               />
             </div>
             <div class="mb-3">
-              <label for="x-axis" class="form-label">Select x-axis Type</label>
```
Comment (ianmuchyri) [inline ui/web/templates/charts/piechartmodal.html:48]: In this chart you need to populate the chart with the form data

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: ui/web/templates/charts/stackedlinechartmodal.html [hunk 4]
```diff
@@ -120,10 +254,16 @@ <h5 class="modal-title" id="stackedLineChartModalLabel">Stacked line charts</h5>
             xAxis: {
               type: 'category',
               boundaryGap: false,
-              data: ['Thing1', 'Thing2', 'Thing3', 'Thing4', 'Thing5', 'Thing6', 'Thing7']
+              data: ['Thing1', 'Thing2', 'Thing3', 'Thing4', 'Thing5', 'Thing6', 'Thing7'],
```
Comment (ianmuchyri) [inline ui/web/templates/charts/stackedlinechartmodal.html:257]: I am not able to see the x-axis label
image

File: ui/web/templates/charts/dynamicdatachartmodal.html [hunk 5]
```diff
@@ -85,7 +202,7 @@ <h5 class="modal-title" id="dynamicDataChartModalLabel">Dynamic Data chart</h5>
     document
       .getElementById("create-dynamicDataChart-button")
       .addEventListener("click", function () {
-        var form = document.getElementById("create-dynamicDataChart-form");
+        var form = document.getElementById("create-dynamicDataChart-appearance-form");
```
Comment (ianmuchyri) [inline ui/web/templates/charts/dynamicdatachartmodal.html:205]: populate the chart appearance with data from the form

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: ui/web/templates/charts/arealinechartmodal.html [hunk 6]
```diff
@@ -22,68 +22,185 @@ <h5 class="modal-title" id="areaLineChartModalLabel">Area line charts</h5>
...
+                  />
+                </div>
+                <div class="mb-3">
+                  <label for="time-interval" class="form-label">Enter the time internal</label>
+                  <input
+                    type="text"
+                    class="form-control mb-3"
+                    name="timeInterval"
+                    id="time-interval"
+                    placeholder="Enter the time interval in seconds"
```
[the remark this answers is not in the document]
Comment (dborovcanin) [reply ui/web/templates/charts/arealinechartmodal.html:138]: Seconds are fine. We can add minutes later if needed.

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Please resolve conflicts.

File: ui/web/templates/charts/dynamicdatachartmodal.html [hunk 7]
```diff
@@ -213,7 +369,7 @@ <h5 class="modal-title" id="dynamicDataChartModalLabel">Dynamic Data chart</h5>
           app.count = 11;
           setInterval(function () {
               let axisData = new Date().toLocaleTimeString().replace(/^\D*/, '');
-              data.shift();
+              data.shift();create-areaLineChart-button
```
Comment (ianmuchyri) [inline ui/web/templates/charts/dynamicdatachartmodal.html:372]:
```suggestion
              data.shift();
```

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**